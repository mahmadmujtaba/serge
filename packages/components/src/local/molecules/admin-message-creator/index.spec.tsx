/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import AdminMessageCreator from './index'

const force = {
  name: 'blue',
  color: '#6699cc',
  icon: ''
}

describe('AdminMessageCreator component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AdminMessageCreator from={force} channel={'Game Admin'} role={'Umpire'} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
