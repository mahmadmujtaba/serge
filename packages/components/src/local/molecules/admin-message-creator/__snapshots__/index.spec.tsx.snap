// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMessageCreator component: renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-1 message-creator__wrap"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <section
      className="input-container"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            disabled={false}
            name="Message"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            rows={4}
            value=""
          />
        </div>
      </div>
    </section>
  </div>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained btn-send MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          aria-disabled={false}
          aria-label="Send"
          className="MuiButtonBase-root MuiIconButton-root"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4.01 6.03l7.51 3.22-7.52-1 .01-2.22m7.5 8.72L4 17.97v-2.22l7.51-1M2.01 3L2 10l15 2-15 2 .01 7L23 12 2.01 3z"
              />
            </svg>
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`MessageCreator renders correctly 1`] = `
<div>
  <section
    className="input-container"
  >
    <h2>
      Post message
    </h2>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          disabled={false}
          name="Message"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          rows={4}
          value=""
        />
      </div>
    </div>
  </section>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Send
    </span>
  </button>
</div>
`;
